### PR TITLE
XW-252 | load optimizely after ember starts

### PR DESCRIPTION
**Pull request by tomasznapieralski**
https://wikia-inc.atlassian.net/browse/XW-252

### CODE REVIEW

File: front/scripts/main/app.ts [hunk 1]
```diff
@@ -59,7 +59,23 @@ App.initializer({
 });
 
 App.initializer({
+	name: 'optimizely',
+	initialize () {
+		App.deferReadiness();
+
+		if (M.prop('optimizelyScript') !== undefined && !M.getQueryParam('noexternals')) {
```
Comment (rogatty) [inline front/scripts/main/app.ts]: Put `M.prop('optimizelyScript')` in a variable and use `!Em.isEmpty(optimizelyScript)`. We don't want to load `null` or empty string.

File: front/scripts/main/app.ts [hunk 2]
```diff
@@ -59,7 +59,23 @@ App.initializer({
...
+			Em.$.getScript(M.prop('optimizelyScript')).then(
```
Comment (rogatty) [inline front/scripts/main/app.ts]: This is weird use of `then`. I think it should be
```
.then(() => {
    App.advanceReadiness();
});
```

File: server/lib/Utils.ts [hunk 3]
```diff
@@ -199,7 +199,11 @@ export function createServerData(localSettings: LocalSettings, wikiDomain: strin
 		mediawikiDomain: getWikiDomainName(localSettings, wikiDomain),
 		apiBase: localSettings.apiBase,
 		environment: getEnvironmentString(env),
-		cdnBaseUrl: getCDNBaseUrl(localSettings)
+		cdnBaseUrl: getCDNBaseUrl(localSettings),
+		optimizelyScript:
+			(localSettings.optimizely.enabled) ?
+			localSettings.optimizely.scriptPath + localSettings.optimizely.account + '.js' :
+			undefined
```
Comment (rogatty) [inline server/lib/Utils.ts]: If you want something to be `undefined` then you shouldn't define it at first place.

File: server/views/_layouts/ember-main.hbs [hunk 4]
```diff
@@ -72,6 +72,9 @@
 			{{#if error}}
 				M.provide('error', {{{error}}});
 			{{/if}}
+			{{#if server.optimizelyScript}}
+                M.prop('optimizelyScript', '{{{server.optimizelyScript}}}');
```
Comment (rogatty) [inline server/views/_layouts/ember-main.hbs]: Wrong indentation.

File: front/scripts/main/app.ts [hunk 5]
```diff
@@ -59,7 +59,23 @@ App.initializer({
 });
 
 App.initializer({
+	name: 'optimizely',
+	initialize () {
+		App.deferReadiness();
```
Comment (bognix) [inline front/scripts/main/app.ts]: can this be done inside if statement?
This way you don't have to advanceRediness inside "else"

File: front/scripts/main/app.ts (hunk 2 above)
Comment (rogatty) [inline front/scripts/main/app.ts]: Also, what if the request fails? Currently Ember won't load. Maybe we should use `.always()` instead?
http://api.jquery.com/jQuery.ajax/#jqXHR

File: front/scripts/main/app.ts [hunk 6]
```diff
@@ -59,7 +59,23 @@ App.initializer({
 });
 
 App.initializer({
+	name: 'optimizely',
+	initialize () {
```
Comment (bognix) [inline front/scripts/main/app.ts]: missing return type

File: front/scripts/main/app.ts [hunk 7]
```diff
@@ -59,7 +59,23 @@ App.initializer({
...
+		var optimizelyScript =  M.prop('optimizelyScript');
```
Comment (rogatty) [inline front/scripts/main/app.ts]: Too... much... space :see_no_evil:

**Comment by rogatty** [comment]
LGTM

**Pull request merged**